pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networkpolicy
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/contiv/libOpenflow/openflow13"
+	"github.com/contiv/libOpenflow/protocol"
+	"github.com/contiv/ofnet/ofctrl"
+	"gopkg.in/natefinch/lumberjack.v2"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
+	opsv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/ops/v1alpha1"
+	binding "github.com/vmware-tanzu/antrea/pkg/ovs/openflow"
+)
+
+const (
+	logDir      string = "/var/log/antrea/networkpolicy/"
+	logfileName string = "np.log"
+)
+
+var (
+	AntreaPolicyLogger *log.Logger
+)
+
+// logInfo will be set by retrieving info from packetin and register
+type logInfo struct {
+	tableName   string // name of the table sending packetin
+	npName      string // namespace/name for Antrea NetworkPolicy and name for Antrea ClusterNetworkPolicy
+	disposition string // Allow/Drop of the rule sending packetin
+	ofPriority  string // openflow priority of the flow sending packetin
+	srcIP       string // source IP of the traffic logged
+	destIP      string // destination IP of the traffic logged
+	pktLength   uint16 // packet length of packetin
+	protocolStr string // protocol of the traffic logged
+}
+
+// initLogger is called while newing Antrea network policy agent controller.
+// Customize AntreaPolicyLogger specifically for Antrea Policies audit logging.
+func initLogger() error {
+	// logging file should be /var/log/antrea/networkpolicy/np.log
+	if _, err := os.Stat(logDir); os.IsNotExist(err) {
+		os.Mkdir(logDir, 0755)
+	}
+	file, err := os.OpenFile(logDir+logfileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
+	if err != nil {
+		klog.Errorf("Failed to initialize logger to audit Antrea Policies %v", err)
+		return err
+	}
+
+	AntreaPolicyLogger = log.New(file, "", log.Ldate|log.Lmicroseconds)
+	// Use lumberjack log file rotation
+	AntreaPolicyLogger.SetOutput(&lumberjack.Logger{
+		Filename:   logDir + logfileName,
+		MaxSize:    500, // megabytes after which new file is created
+		MaxBackups: 3,   // number of backups
+		MaxAge:     28,  //days
+		Compress:   true,
```

Comment:
@jianjuns i believe this PR does compress the log files as is done here..

Reply:
@abhiraut: I meant whether or not to use human readable values/names in the log messages, or just some binary format. Do you know how other solutions/products log?

Reply:
i noticed Cilium uses the exact same thing

Reply:
are you suggesting we log IDs instead of names? for example instead of table Name use its ID... protocol -> use a number?

Reply:
I would be in favor of starting with the human-readable format (with key-value entries) so that we can read the logs easily and integrate with log collectors. We can add a binary format later for export if we want for integration with certain tools, and also provide functionality in antctl to convert between formats at that time.

Reply:
@abhiraut and I talked offline, and seems other solutions we know also use readable format. So, I am fine with that.